notebooks/Threshold Optimization Postprocessing for Binary Classification.ipynb

```diff
@@ -0,0 +1,458 @@
+    "\n",
+    "$$\n",
+    "P[\\text{recidivism predicted} | \\text{African-American, recidivism}] = 0.95 \\neq 0.15 = P[\\text{recidivism predicted} | \\text{African-American, no ecidivism}]\n",
+    "$$\n",
+    "\n",
+    "Assessing which disparity metric is indeed fair varies by application scenario."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# This wrapper around the unconstrained estimator serves the purpose of mapping the predict\n",
+    "# method to predict_proba so that we can use real values to get more accurate estimates.\n",
+    "class LogisticRegressionAsRegression:\n",
+    "    def __init__(self, logistic_regression_estimator):\n",
+    "        self.logistic_regression_estimator = logistic_regression_estimator\n",
+    "    \n",
+    "    def fit(self, X, y):\n",
+    "        self.logistic_regression_estimator.fit(X, y)\n",
+    "    \n",
+    "    def predict(self, X):\n",
+    "        # use predict_proba to get real values instead of 0/1, select only prob for 1\n",
+    "        scores = self.logistic_regression_estimator.predict_proba(X)[:,1]\n",
+    "        return scores"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from fairlearn.postprocessing import ThresholdOptimizer\n",
+    "from copy import deepcopy\n",
+    "\n",
+    "unconstrained_predictor_wrapper = LogisticRegressionAsRegression(unconstrained_predictor)\n",
+    "postprocessed_predictor_EO = ThresholdOptimizer(\n",
+    "    unconstrained_predictor=unconstrained_predictor_wrapper,\n",
+    "    constraints=\"equalized_odds\",\n",
+    "    plot=True)\n",
+    "\n",
+    "postprocessed_predictor_EO.fit(X_train, y_train, sensitive_features=sensitive_features_train)\n",
+    "\n",
+    "fairness_aware_predictions_EO_train = postprocessed_predictor_EO.predict(X_train, sensitive_features=sensitive_features_train)\n",
+    "fairness_aware_predictions_EO_test = postprocessed_predictor_EO.predict(X_test, sensitive_features=sensitive_features_test)\n",
+    "\n",
+    "show_proportions(X_train, sensitive_features_train, fairness_aware_predictions_EO_train, y_train, description=\"equalized odds with postprocessed model on training data:\", plot_row_index=1)\n",
+    "show_proportions(X_test, sensitive_features_test, fairness_aware_predictions_EO_test, y_test, description=\"equalized odds with postprocessed model on test data:\", plot_row_index=2)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predicted_ys = [\n",
+    "    unconstrained_predictor.predict(X_test).tolist(),\n",
+    "    postprocessed_predictor_EO.predict(X_test, sensitive_features=sensitive_features_test).tolist()\n",
+    "]\n",
+    "\n",
+    "FairlearnDashboard(sensitive_features=sensitive_features_mapped,\n",
```

Comment:
What am I supposed to do and see with this dashboard?

Reply:
If I try looking at the accuracy plot, I turn two options - high disparity and lower accuracy, and lower disparity and higher accuracy. This seems odd.

Reply:
The 'recall score' plot makes more sense, with lower recall paired with lower disparity.

Reply:
Oh it shows the disparity in accuracy instead of disparity in predictions. Weird, how can I change that?

Reply:
This will probably be a bigger discussion on how to address this, outside of scope for this PR